lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -217,7 +220,9 @@ def update_action_builder(args, source)
         end
         source['script']['lang'] = @options[:script_lang] if @options[:script_lang] != ''
       else
-        source = { 'doc' => source }
+        unless source.has_key?('doc')
```

Comment:
@salimane, have you read my comments in #356 and @jordansissel's comment in #357 ?
Here, 'doc' could be a legitimate user's field !